src/EditModel/event-program/data-entry-form/EditCustomFormProgramStage.js

```diff
@@ -146,20 +121,14 @@ class EditDataEntryForm extends React.Component {
         };
 
         this.disposables = new Set();
-
-        // Load form data, operands and flags
-        this.disposables.add(Observable.combineLatest(
-            firstProgramStage$,
-            programStageDataElements$,
-            Observable.fromPromise(context.d2.Api.getApi().get('system/flags')),
-            (programStage, ops, flags) => ([programStage, ops, flags])
-        )
-            .take(1)
-            .subscribe(([programStage, programStageDataElements, flags]) => {
+        const { programStage, programStageDataElements } = props;
+            // Load flags
+            this.disposables.add(Observable.fromPromise(context.d2.Api.getApi().get('system/flags'))
+            .subscribe(flags => {
```

Comment:
This function should be split up into smaller ones, right now it has many responsibilities (flags, operands, creating inserter functions, filtering elements, etc, that do not need to be inside the `next` callback.

I think only these two things in the function actually use the parameter:

```
[..].subscribe(flags => {
                 this.flags = flags.reduce((out, flag) => {
                     out[flag.path] = flag.name; // eslint-disable-line
                     return out;
                 }, {});

                 Object.keys(this.flags).forEach((flag) => {
                     insertFn[flag] = this.insertFlag.bind(this, flag);
                 });
});
```

The inserter functions could also be abstracted away to a function like `inserter(...)` which could be reused for operands and flags, making it a bit simpler as well.

I like functions which do one thing. I think they are a lot easier to reason about. This functions does many things. :)